product_brand/product_brand.py

```diff
@@ -1,55 +1,66 @@
-# You should have received a copy of the GNU Affero General Public License #  #
-# along with this program. If not, see <http://www.gnu.org/licenses/>. #      #
-# #                                                                           #
+# Leonardo Donelli, donelli@webmonks.it, http://www.wearemonk.com             #
+#                                                                             #
+# This program is free software: you can redistribute it and/or modify        #
+# it under the terms of the GNU Affero General Public License as              #
+# published by the Free Software Foundation, either version 3 of the          #
+# License, or (at your option) any later version.                             #
+#                                                                             #
+# This program is distributed in the hope that it will be useful,             #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of              #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                #
+# GNU Affero General Public License for more details.                         #
+#                                                                             #
+# You should have received a copy of the GNU Affero General Public License    #
+# along with this program. If not, see <http://www.gnu.org/licenses/>.        #
+#                                                                             #
 ###############################################################################
 ###############################################################################
 # Product Brand is an Openobject module wich enable Brand management for      #
 # products                                                                    #
 ###############################################################################
-from openerp.osv import orm, fields
+from openerp import models, fields
 
 
-class product_brand(orm.Model):
+class ProductBrand(models.Model):
     _name = 'product.brand'
-    _columns = {
-        'name': fields.char('Brand Name'),
-        'description': fields.text('Description', translate=True),
-        'partner_id': fields.many2one(
-            'res.partner', 'partner',
-            help='Select a partner for this brand if it exists.',
-            ondelete='restrict'
-        ),
-        'logo': fields.binary('Logo File'),
-    }
 
+    name = fields.Char('Brand Name')
+    description = fields.Text('Description', translate=True)
+    partner_id = fields.Many2one(
+        'res.partner',
+        string='Partner',
+        help='Select a partner for this brand if it exists',
+        ondelete='restrict'
+    )
+    logo = fields.Binary('Logo File')
+    products_count = fields.Integer(
+        string='Number of products',
+        compute='_get_products_count',
+    )
 
-class product_template(orm.Model):
+    def _get_products_count(self):
```

Comment:
Use `@api.depends` over the recently created one2many, and then you need only to make `len()` over that field.

Use also @api.one and you will save a line.